src/Modules/Guilds/Hooks/useVotingPowerForProposalExecution.ts

```diff
@@ -0,0 +1,93 @@
+import { useMemo } from 'react';
+import { BigNumber } from 'ethers';
+import { useContractRead } from 'wagmi';
+import useGuildImplementationType from 'Modules/Guilds/Hooks/useGuildImplementationType';
+import { WagmiUseContractReadResponse } from 'Modules/Guilds/Hooks/types';
+import useSnapshotId from 'Modules/Guilds/Hooks/useSnapshotId';
+import { useTypedParams } from 'Modules/Guilds/Hooks/useTypedParams';
+import ERC20GuildContract from 'contracts/ERC20Guild.json';
+import SnapshotERC20Guild from 'contracts/SnapshotERC20Guild.json';
+import SnapshotRepERC20Guild from 'contracts/SnapshotRepERC20Guild.json';
+
+/**
+ * -  ERC20GuildContract
+ *    function getVotingPowerForProposalExecution() public view virtual returns (uint256) {}
+ *
+ * -  SnapshotERC20Guild:
+ *    function getVotingPowerForProposalExecution(uint256 snapshotId) public view virtual returns (uint256) {}
+ *
+ * -  SnapshotRepERC20Guild
+ *    function getSnapshotVotingPowerForProposalExecution(bytes32 proposalId) public view virtual returns (uint256) {}
+ */
+
+interface UseVotingPowerForProposalExecutionProps {
+  contractAddress: string;
+  proposalId?: string;
+}
+
+const getConfig = ({

```

Comment:
🔤 Typescript this?

Reply:
Sure, added arg types. Not return type since that should be consumed from wagmi. I don't think we should continue expanding our codebase with our own types for external libs.